src/documents/ContextAwareness.md

```diff
@@ -0,0 +1,327 @@
+        </tr>
+        <tr>
+            <th>Member</th>
+            <th>Type</th>
+            <th>Description</th>
+        </tr>
+    </thead>
+    <tbody>
+        <tr>
+            <td><code>func</code>/<code>funcName</code></td>
+            <td><code>String</code> <a href="IoCReferences.md">IoC reference</a></td>
+            <td>A function or function name to be evaluated to produce the context value</td>
+        </tr>
+        <tr>
+            <td><code>value</code></td>
+            <td><code>String</code>, <code>Number</code> or <code>Boolean</code></td>
+            <td>The value to be supplied at the <code>value</code> path in the options tructure of the constructed context</td>
+        </tr>
+
+    </tbody>
+</table>
+
+You must supply exactly one of `func`, `funcName` or `value`.
+
+### Example of `fluid.contextAware.makeChecks`
+
+```
+fluid.contextAware.makeChecks({
+    "fluid.browser.supportsBinaryXHR": {
+        funcName: "fluid.enhance.supportsBinaryXHR"
+    },
+    "fluid.browser.supportsFormData": {
+        funcName: "fluid.enhance.supportsFormData"
+    }
+});
+```
+
+In the above example, the two global functions `fluid.enhance.supportsBinaryXHR` and `fluid.enhance.supportsFormData` will be executed, and their
+return values added into contexts with the names `fluid.browser.supportsBinaryXHR` and `fluid.browser.supportsFormData`.
+
+The contexts registered by ``fluid.contextAware.makeChecks`` can be erased from the system by the use of the call 
+
+```
+fluid.contextAware.forgetChecks(`contextNames`);
+```
+
+Here, `contextNames` is can hold either a `String` or `Array of String` holding the keys from the structures previously supplied to `fluid.contextAware.makeChecks`
+
+### Example of `fluid.contextAware.forgetChecks`
+
+For example, the checks registered in the above example `fluid.contextAware.makeChecks` call could be erased by a call to
+
+```
+fluid.contextAware.forgetChecks(["fluid.browser.supportsBinaryXHR", "fluid.browser.supportsFormData"]);
+```
+
+
+## Defining and broadcasting a fresh adaptation in one operation with `fluid.contextAware.makeAdaptation`
+
+A very common use case is to define an adaptation (that is, a `distributeOptions` block which targets the `contextAwareness` area of a collection
+of componentsin the tree), and then to create an instance of a single, well-known component which actually broadcasts the adaptation. This was
+what we did in two steps (`fluid.defaults` plus `fluid.constructSingle`) in the above [example `contextAwareness` broadcast](#example-of-dynamically-broadcasting-a-fresh-adaptation)
+can be done in a single step using the `fluid.contextAware.makeAdaptation` API. 
+
+```
+fluid.contextAware.makeAdaptation(<adaptationRecord>);
+```
+
+This accepts a single complex structure `adaptationRecord` with a number of fields that need to be filled in:
```

Comment:
1) No need for the word 'complex' here
2) Perhaps instead of "a number of fields that need to be filled in" maybe "a number of required fields" ?